### ISSUE TITLE
Black background when converting SVG to PNG

### ISSUE THREAD

**Issue by giancarlobastos** [srwiley/oksvg#34]
I'm trying to convert a `svg` file to `png`, but the output file ended with a black background instead of white.

```
package converter

import (
	"image"
	"image/png"

	"net/http"
	"os"

	"github.com/srwiley/oksvg"
	"github.com/srwiley/rasterx"
)

func ConvertSvgToPng() error {
	url := "https://s.glbimg.com/es/sde/f/organizacoes/2020/02/12/botsvg.svg"
	w, h := 512, 512

	response, err := http.Get(url)
	if err != nil {
		return err
	}
	defer response.Body.Close()

	icon, _ := oksvg.ReadIconStream(response.Body)
	icon.SetTarget(0, 0, float64(w), float64(h))

	rgba := image.NewRGBA(image.Rect(0, 0, w, h))
	icon.Draw(rasterx.NewDasher(w, h, rasterx.NewScannerGV(w, h, rgba, rgba.Bounds())), 1)

	out, err := os.Create("out.png")
	if err != nil {
		return err
	}
	defer out.Close()

	err = png.Encode(out, rgba)
	if err != nil {
		return err
	}
	return err
}
```
SVG file: [https://s.glbimg.com/es/sde/f/organizacoes/2020/02/12/botsvg.svg](https://s.glbimg.com/es/sde/f/organizacoes/2020/02/12/botsvg.svg)

PNG output:
![out](https://user-images.githubusercontent.com/4674138/142613032-825d100a-b5db-489d-b9e7-d4cfef9722d7.png)

**Comment by mysugar** [comment]
Hi ,have you solved this problem? @giancarlobastos

**Comment by yeldiRium** [comment]
Some input on this: I believe this is due to oksvg not supporting `style` tags in SVG. I can't say for sure, but I'm having the same problem and removing the `style` block and putting the style attributes directly into the `path` tags solved the rendering issue for me.

**Comment by yeldiRium** [comment]
I've perused the code and I've found something that confuses me:

`style` tag should be handled here https://github.com/srwiley/oksvg/blob/a61f04f16b769b01d0ff6df7596ab1464f3cf630/public.go#L61 and the code to do so looks fine to me. Once the `endElement` for the `style` tag is found, its text content is parsed and added to the styles here https://github.com/srwiley/oksvg/blob/a61f04f16b769b01d0ff6df7596ab1464f3cf630/public.go#L89.

But before the `style` tag can be handled, the `readStartElement` call here https://github.com/srwiley/oksvg/blob/a61f04f16b769b01d0ff6df7596ab1464f3cf630/public.go#L57 will always fail, since `style` tags are not handled there and not a valid draw function, so this line will fail https://github.com/srwiley/oksvg/blob/a61f04f16b769b01d0ff6df7596ab1464f3cf630/icon_cursor.go#L345.

I'm not that far into the code, so I'm not intuitively sure enough on how to fix this to provide a PR. However, my intuition is to handle `style` tags before and without calling `readStartElement` on them. @srwiley what do you think about this? Can you do anything with this research?

**Comment by yeldiRium** [comment]
One more update for me (sorry for the multipost):

After looking at the tests I realize that the `style` tag _is_ supported, but only within an enclosing `defs` tag. That is understandable, but is probably not how most SVGs look and as far as I know there is nothing in the SVG spec that says that `style` tags need to be enclosed in `defs` tags. @srwiley would you be fine with changing this requirement?

**Comment by srwiley** [comment]
Yes, I see no reason why that should be mandatory if it is not in the
specifications.

**Comment by yeldiRium** [comment]
That is great to hear. Would you mind taking a look at [my PR](https://github.com/srwiley/oksvg/pull/40) then?
I'm not sure how clean my implementation is, since I basically circumvent the `inDefs` check by setting `inDefsStyle` without it. But I added a test and it does what it's supposed to do.

**Comment by srwiley** [comment]
Yes, is everyone in agreement? Have the tests been run and are there tests
supporting the new code?

**Comment by goloroden** [comment]
Yes, there are tests to support the code (and given the discussion above it seems as if @yeldiRium did a pretty decent job 😊).
